source/config/Config.cpp

```diff
@@ -433,6 +433,82 @@ bool PlainConfig::Validate() const
     return true;
 }
 
+void PlainConfig::SerializeToObject(Crt::JsonObject &object) const
+{
+    if (endpoint.has_value() && endpoint->c_str())
+    {
+        object.WithString(JSON_KEY_ENDPOINT, endpoint->c_str());
+    }
+    if (cert.has_value() && cert->c_str())
+    {
+        object.WithString(JSON_KEY_CERT, cert->c_str());
+    }
+    if (key.has_value() && key->c_str())
+    {
+        object.WithString(JSON_KEY_KEY, key->c_str());
+    }
+    if (rootCa.has_value() && rootCa->c_str())
+    {
+        object.WithString(JSON_KEY_ROOT_CA, rootCa->c_str());
+    }
+    if (thingName.has_value() && thingName->c_str())
+    {
+        object.WithString(JSON_KEY_THING_NAME, thingName->c_str());
+    }
+
+    JsonObject loggingObject;
+    logConfig.SerializeToObject(loggingObject);
+    object.WithObject(JSON_KEY_LOGGING, loggingObject);
+
+    JsonObject jobsObject;
+    jobs.SerializeToObject(jobsObject);
+    object.WithObject(JSON_KEY_JOBS, jobsObject);
+
+    JsonObject tunnelingObject;
+    tunneling.SerializeToObject(tunnelingObject);
+    object.WithObject(JSON_KEY_TUNNELING, tunnelingObject);
+
+    JsonObject deviceDefenderObject;
+    deviceDefender.SerializeToObject(deviceDefenderObject);
+    object.WithObject(JSON_KEY_DEVICE_DEFENDER, deviceDefenderObject);
+
+    JsonObject fleetProvisioningObject;
+    fleetProvisioning.SerializeToObject(fleetProvisioningObject);
+    object.WithObject(JSON_KEY_FLEET_PROVISIONING, fleetProvisioningObject);
+
+    if (fleetProvisioning.enabled)
```

Comment:
Since we said we would serialize features that aren't enabled, doesn't that mean we need to remove this check?

Reply:
yup, this was left in from earlier

Reply:
Resolved offline, will keep the check